packages/ui/src/components/va-rating/VaRating.vue

```diff
@@ -1,208 +1,101 @@
+        v-for="itemNumber in $props.max"
+        v-bind="VaRatingItemProps"
+        :model-value="getItemValue(itemNumber - 1)"
+        :key="itemNumber"
+        :empty-icon-color="$props.unselectedColor"
+        :hover="isHoverEnabled"
+        :tabindex="tabindex"
+        @hover="onItemHoveredValueUpdate(itemNumber - 1, $event)"
+        @update:model-value="isInteractionsEnabled && onItemValueUpdate(itemNumber - 1, $event)"
       >
-        <va-rating-item
-          v-for="number in $props.max"
-          :key="number"
-          class="va-rating__number-item"
-          :halves="$props.halves"
-          :hover="hoverEnabled"
-          :size="$props.size"
-          :color="$props.color"
-          :empty-icon-color="$props.unselectedColor"
-          :tabindex="tabIndex"
-          :modelValue="getItemValue(number)"
-          @hover="onHover(number, $event)"
-          @click="onItemSelected(number, $event)"
-        >
-          <template v-slot="{ props }">
-            <button
-              @click="props.onClick"
-              class="va-rating__number-item"
-              tabindex="-1"
-              :style=" {
-                background: props.value === 0.5
-                  ? `linear-gradient(90deg, ${colorComputed} 50%, ${focusColor} 50%`
-                  : !props.value ? focusColor : colorComputed,
-                color: props.value ? '#fff' : colorComputed,
-                width: sizeComputed,
-                height: sizeComputed,
-                fontSize: fontSizeComputed,
-                borderRadius: `${SizeMixin_fontSize * 0.125}rem`,
-              }"
-            > {{ number }} </button>
-          </template>
-        </va-rating-item>
-      </template>
-      <template v-else>
-        <va-rating-item
-          v-for="itemNumber in $props.max"
-          :key="itemNumber"
-          :halves="$props.halves"
-          :hover="hoverEnabled"
-          :filled-icon-name="$props.icon"
-          :half-icon-name="$props.halfIcon"
-          :empty-icon-name="$props.emptyIcon"
-          :size="$props.size"
-          :color="$props.color"
-          :empty-icon-color="$props.unselectedColor"
-          :tabindex="tabIndex"
-          :modelValue="getItemValue(itemNumber)"
-          @hover="onHover(itemNumber, $event)"
-          @click="onItemSelected(itemNumber, $event)"
-        />
-      </template>
+        <template v-if="$props.numbers" v-slot="{ props }">
+          <VaRatingItemNumberButton
+            v-bind="VaRatingItemNumberButtonProps"
+            :model-value="props.value"
+            :item-number="itemNumber"
+            @click="props.onClick"
+          />
+        </template>
+      </va-rating-item>
     </div>
-    <span class="va-rating__text-wrapper" v-if="$props.texts.length === $props.max" :style="{ color: computeColor($props.textColor) }">
-      {{ $props.texts[Math.round(valueProxy) - 1] }}
+    <span
+      v-if="$props.texts && $props.texts.length === $props.max"
```

Comment:
$props.texts && 
```suggestion
      v-if="$props.texts.length === $props.max"
```
props.texts has array as default value so it's redundant

Reply:
I have ts error here. Vue think even if there is default in prop definition, it can be undefined. This behavior caused because of some technically issue inside vue.